service/storage_service.cc

```diff
@@ -2184,12 +2185,13 @@ future<> storage_service::track_upgrade_progress_to_topology_coordinator(sharded
     }
 }
 
-std::list<locator::host_id_or_endpoint> storage_service::parse_node_list(sstring comma_separated_list) {
-    std::vector<sstring> ignore_nodes_strs = utils::split_comma_separated_list(std::move(comma_separated_list));
-    std::list<locator::host_id_or_endpoint> ignore_nodes;
+std::vector<locator::host_id_or_endpoint> storage_service::parse_node_list(const sstring& comma_separated_list) {
+    std::vector<sstring> ignore_nodes_strs = utils::split_comma_separated_list(comma_separated_list);
+    locator::host_id_or_endpoint_array ignore_nodes;
+    ignore_nodes.reserve(ignore_nodes_strs.size());
     for (const sstring& n : ignore_nodes_strs) {
```

Comment:
This looks similar to `string_list_to_enpoint_array` except the other doesn't have error checking. Weird.

Reply:
Removed duplication.